### PR TITLE
feat(query): add ast counterpart to ObjectLiteral

### PR DESCRIPTION
**Pull request by u9g**

### CODE REVIEW

File: crates/oxc_query/src/schema.graphql [hunk 1]
```diff
@@ -306,6 +306,18 @@ type ObjectLiteral implements Expression & HasSpan {
...
 
+type ObjectLiteralAST implements Expression & HasSpan & ASTNode {
+  value(key: String!): [Expression!]!
+
+  # Expression
+  as_constant_string: String
+  # ASTNode
+  parent: ASTNode
+  ancestor: [ASTNode!]!
+  # HasSpan
```
Comment (DonIsaac) [inline crates/oxc_query/src/schema.graphql:317]: maybe provide a more useful message? Maybe even a doc comment?

Comment (u9g) [reply crates/oxc_query/src/schema.graphql:317]: yes, but in a future pr

File: crates/oxc_query/src/adapter.rs [hunk 2]
```diff
@@ -136,11 +136,13 @@ impl<'a, 'b: 'a> trustfall::provider::Adapter<'a> for &'a Adapter<'b> {
                 property_name.as_ref(),
                 resolve_info,
             ),
-            "ObjectLiteral" => super::properties::resolve_object_literal_property(
-                contexts,
-                property_name.as_ref(),
-                resolve_info,
-            ),
+            "ObjectLiteralAST" | "ObjectLiteral" => {
```
Comment (DonIsaac) [inline crates/oxc_query/src/adapter.rs:139]: What's the difference between these two? 

Separately, why are we using strings here instead of enums?

**Review by DonIsaac — APPROVED** [review]

File: crates/oxc_query/src/adapter.rs (hunk 2 above)
Comment (u9g) [reply crates/oxc_query/src/adapter.rs:139]: Difference: they are different items in the schema.graphql

why strings? That’s what trustfall’s api is.

**Pull request merged**